packages/vm/src/context.rs

```diff
@@ -86,6 +88,13 @@ fn get_context_data<'a, 'b, S: Storage, Q: Querier>(
     Box::leak(owned) // give up ownership
 }
 
+fn get_context_data<'a, 'b, S: Storage, Q: Querier>(ctx: &'a Ctx) -> &'b ContextData<'b, S, Q> {
+    let owned = unsafe {
+        Box::from_raw(ctx.data as *mut ContextData<S, Q>) // obtain ownership
+    };
+    Box::leak(owned) // give up ownership
+}
+
```

Comment:
I think that `get_context_data_mut` can be implemented by just forwarding to this function.
Its a subtlety, but it's nice that you thought of adding this function. You only use it in one place, but i think it's appropriate.

Reply:
I was thinking about that too but was not sure in which direction to forward the call. When `get_context_data_mut` calls this, we get a non-mutable result reference, right?

Reply:
oh yeah, that's right - good point. i didn't try it locally, so i didn't know for sure. calling one from the other wont work

Reply:
a mutable reference can be implicitly cast to a shared reference, but not the other way around.